File: rust/cat.rs
use std::io;
use std::process::ExitCode;

fn main() -> ExitCode {
    let copied: io::Result::<u64> = io::copy(&mut io::stdin(), &mut io::stdout());
    if let Err(error) = copied {
        eprintln!("Failed to copy input to output: {}", error);
        return ExitCode::FAILURE;
    }
    ExitCode::SUCCESS
}

File: csharp/cat.cs
using System;

public class main {
    public static void Main(String[] args) {
        System.IO.Stream stdin = Console.OpenStandardInput();
        System.IO.Stream stdout = Console.OpenStandardOutput();
        try {
            stdin.CopyTo(stdout);
        } catch(Exception e) {
            Console.Error.WriteLine("Failed to copy input to output: " + e.Message);
        }
    }
}
